spec/support/dominos/admin/page.rb

```diff
@@ -21,11 +21,27 @@ def self.sign_in(options)
       def self.sign_in_as(role, options = {})
         email = "#{role}@example.com"
         password = '!Pass123'
+        on = if role == :account_manager && options[:on].blank?
+               FactoryGirl.create(:account)
+             else
+               options.delete(:on)
+             end
+
+        if ['previewer', 'editor', 'publisher', 'manager'].include?(role)
+          user = FactoryGirl.create(:user, options.reverse_merge(email: email,
+                                                                 password: password))
+        else
+          user = FactoryGirl.create(:user, role, options.reverse_merge(email: email,
+                                                                       password: password))
+        end
 
-        user = FactoryGirl.create(:user, role, options.reverse_merge(:email => email, :password => password))
+        unless role == :editor || on.blank?
```

Comment:
`unless` with further logical operators are hard to understand. Prefer `if`. Why is editor not allowed here?